File: warehouse/src/service/storagerackService.js


import axios from 'axios'

export async function getAllStoragerack() {
    let result = await axios.get('http://localhost:8181/backEndWarehouse/storagerack/findAllStoragerack')
    return result.data.data;
}

export async function addStoragerack(storagerack){
    let result = await axios.post('http://localhost:8181/backEndWarehouse/storagerack/addStoragerack', storagerack);
    return result.data;
}

export async function editStoragerack(storagerack){
    let result = await axios.post('http://localhost:8181/backEndWarehouse/storagerack/editStoragerack',storagerack)
    return result.data;
}

export async function deleteStoragerack(srId) {
    let result = await axios.delete('http://localhost:8181/backEndWarehouse/storagerack/deleteStoragerack/'+srId);
    return result.data;
}

export async function getStoragerackBySrRaId(raId) {
    let result = await axios.get('http://localhost:8181/backEndWarehouse/storagerack/getStoragerackBySrRaId',{
        params:{
            raId,
        }
    })
    return result.data;
}

export async function updateMaterial(mtId,mtName,srId){
    let result = await axios.get('http://localhost:8181/backEndWarehouse/storagerack/updateMaterial',{
        params:{
            mtId,
            mtName,
            srId,
        }
    })
    return result.data;
}

export async function clearMaterial(srId){
    let result = await axios.get('http://localhost:8181/backEndWarehouse/storagerack/clearMaterial',{
        params:{
            srId
        }
    })
    return result.data;
}

File: backEndWarehouse/src/main/java/com/managementsystem/service/StoragerackService.java
package com.managementsystem.service;

import com.managementsystem.entity.Storagerack;
import com.baomidou.mybatisplus.extension.service.IService;
import org.apache.ibatis.annotations.Param;

import java.util.List;


public interface StoragerackService extends IService<Storagerack> {

    List<Storagerack> list();

    int addStoragerack(Storagerack storagerack);

    int editStoragerack(Storagerack storagerack);

    int deleteStoragerack(String srId);

    List<Storagerack> getStoragerackBySrRaId(String raId);

    int updateMaterial(String mtId,String mtName, String srId);

    int clearMaterial(String srId);
}
